core/database.py

```
@@ -408,31 +431,55 @@ async def get_access_control_users() -> Set[str]:
-            """
-            INSERT INTO access_control_users (user_id, added_by, note, created_at)
-            VALUES ($1, $2, $3, CURRENT_TIMESTAMP)
-            ON CONFLICT (user_id) DO UPDATE SET
-                added_by = EXCLUDED.added_by,
-                note = EXCLUDED.note;
-            """,
-            user_id,
-            added_by,
-            note,
-        )
+        logger.warning(f"Cannot add access control user {user_id}: database pool not initialized")
```

Ruff (0.15.2)

[warning] Logging statement uses f-string

(G004)